engine/src/main/java/org/terasology/input/InputSystem.java

```diff
@@ -142,254 +113,83 @@ public void setControllerDevice(ControllerDevice controllerDevice) {
-        BindableButtonImpl bindInfo = buttonLookup.get(bindId);
-        controllerBinds.put(button, bindInfo);
-    }
-
-    /**
-     * Enumerates all active input bindings for a given binding.
-     * @param bindId the ID
-     * @return a list of keyboard/mouse inputs that trigger the binding.
-     */
-    public List<Input> getInputsForBindButton(SimpleUri bindId) {
-        List<Input> inputs = new ArrayList<>();
-        for (Entry<Integer, BindableButtonImpl> entry : keyBinds.entrySet()) {
-            if (entry.getValue().getId().equals(bindId)) {
-                inputs.add(InputType.KEY.getInput(entry.getKey()));
-            }
-        }
-
-        for (Entry<MouseInput, BindableButtonImpl> entry : mouseButtonBinds.entrySet()) {
-            if (entry.getValue().getId().equals(bindId)) {
-                inputs.add(entry.getKey());
-            }
-        }
-
-        if (mouseWheelUpBind.getId().equals(bindId)) {
-            inputs.add(MouseInput.WHEEL_UP);
-        }
-
-        if (mouseWheelDownBind.getId().equals(bindId)) {
-            inputs.add(MouseInput.WHEEL_DOWN);
-        }
-
-        return inputs;
-    }
-
-    public BindableAxis registerBindAxis(String id, BindableButton positiveButton, BindableButton negativeButton) {
-        return registerBindAxis(id, new BindAxisEvent(), positiveButton, negativeButton);
-    }
-
-    public BindableAxis registerBindAxis(String id, BindAxisEvent event, SimpleUri positiveButtonId, SimpleUri negativeButtonId) {
-        return registerBindAxis(id, event, getBindButton(positiveButtonId), getBindButton(negativeButtonId));
-    }
-
-    public BindableAxis registerBindAxis(String id, BindAxisEvent event, BindableButton positiveButton, BindableButton negativeButton) {
-        BindableAxisImpl axis = new BindableAxisImpl(id, event, positiveButton, negativeButton);
-        axisBinds.add(axis);
-        return axis;
-    }
-
-    public BindableAxis registerRealBindAxis(String id, BindAxisEvent event) {
-        BindableRealAxis axis = new BindableRealAxis(id.toString(), event);
-        axisBinds.add(axis);
-        axisLookup.put(id, axis);
-        return axis;
+        bindsManager.registerBinds();
     }
 
     public void update(float delta) {
+        updateInputEntities();
         processMouseInput(delta);
         processKeyboardInput(delta);
         processControllerInput(delta);
         processBindRepeats(delta);
         processBindAxis(delta);
     }
 
-    public boolean isCapturingMouse() {
-        return capturingMouse && display.hasFocus();
-    }
-
-    public void setCapturingMouse(boolean capturingMouse) {
-        this.capturingMouse = capturingMouse;
+    private void updateInputEntities() {
+        inputEntities = new EntityRef[] {localPlayer.getClientEntity(), localPlayer.getCharacterEntity()};
     }
 
     private void processMouseInput(float delta) {
-        if (!isCapturingMouse()) {
+        if (!display.hasFocus()) {
             return;
```

Comment:
I think that extracting the `display.hasFocus()` call to a `inputSystem.isCapturingMouse()` method is a better idea than simply calling `hasFocus` everywhere as it makes the intent much more clear.